sormas-backend/src/main/java/de/symeda/sormas/backend/report/WeeklyReportFacadeEjb.java

```diff
@@ -377,9 +377,8 @@ public void generateSubmitWeeklyReportTasks() {
 				// task
 				continue;
 			} else {
-				TaskCriteria pendingUserTaskCriteria =
-					new TaskCriteria().taskType(TaskType.WEEKLY_REPORT_GENERATION).assigneeUser(user.toReference()).taskStatus(TaskStatus.PENDING);
-				List<Task> existingTasks = taskService.findBy(pendingUserTaskCriteria, true);
+				List<Task> existingTasks = taskService
```

Comment:
I would extend the criteria to accept multiple statuses but if you would like to keep the new method please rename to `findByTypeAndStatus` or similar and maybe use rest params for statuses (eg. TaskStatus ...status )

Reply:
First thing I did was extending the criteria. The problem with this is that it's used by the task grid, and therefore requires the one status attribute. Having both (one and multiple statuses) seemed ugly to me.
So I will keep the new method, but will adapt the to your requests.